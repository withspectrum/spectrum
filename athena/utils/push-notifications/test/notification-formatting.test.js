// @flow
import formatNotification from '../notification-formatting';
import CHANNEL_CREATED_FIXTURE from './fixtures/CHANNEL_CREATED.json';
import COMMUNITY_INVITE_FIXTURE from './fixtures/COMMUNITY_INVITE.json';
import MESSAGE_CREATED_FIXTURE from './fixtures/MESSAGE_CREATED.json';
import DIRECT_MESSAGE_CREATED_FIXTURE from './fixtures/DIRECT_MESSAGE_CREATED.json';
import REACTION_CREATED_FIXTURE from './fixtures/REACTION_CREATED.json';
import THREAD_CREATED_FIXTURE from './fixtures/THREAD_CREATED.json';
import USER_JOINED_COMMUNITY_FIXTURE from './fixtures/USER_JOINED_COMMUNITY.json';
import MENTION_MESSAGE_FIXTURE from './fixtures/MENTION_MESSAGE';
import MENTION_THREAD_FIXTURE from './fixtures/MENTION_THREAD';

const USER_ID = 'gVk5mYwccUOEKiN5vtOouqroGKo1';

it('should format a channel creation notification', () => {
  let result = formatNotification(CHANNEL_CREATED_FIXTURE, USER_ID);
  expect(result.raw).toBeDefined();
  delete result.raw; // Don't need raw data
  expect(result).toMatchSnapshot();
});

it('should format a community invite notification', () => {
  let result = formatNotification(COMMUNITY_INVITE_FIXTURE, USER_ID);
  expect(result.raw).toBeDefined();
  delete result.raw; // Don't need raw data
  expect(result).toMatchSnapshot();
});

it('should format a message creation notification', () => {
  let result = formatNotification(MESSAGE_CREATED_FIXTURE, USER_ID);
  expect(result.raw).toBeDefined();
  delete result.raw; // Don't need raw data
  expect(result).toMatchSnapshot();
});

it('should format a direct message creation notification', () => {
  let result = formatNotification(DIRECT_MESSAGE_CREATED_FIXTURE, USER_ID);
  expect(result.raw).toBeDefined();
  delete result.raw; // Don't need raw data
  expect(result).toMatchSnapshot();
});

it('should format a reaction creation notification', () => {
  let result = formatNotification(REACTION_CREATED_FIXTURE, USER_ID);
  expect(result.raw).toBeDefined();
  delete result.raw; // Don't need raw data
  expect(result).toMatchSnapshot();
});

it('should format a thread creation notification', () => {
  let result = formatNotification(THREAD_CREATED_FIXTURE, USER_ID);
  expect(result.raw).toBeDefined();
  delete result.raw; // Don't need raw data
  expect(result).toMatchSnapshot();
});

it('should format a user joined community notification', () => {
  let result = formatNotification(USER_JOINED_COMMUNITY_FIXTURE, USER_ID);
  expect(result.raw).toBeDefined();
  delete result.raw; // Don't need raw data
  expect(result).toMatchSnapshot();
});

it('should format a message mention notification', () => {
  let result = formatNotification(MENTION_MESSAGE_FIXTURE, USER_ID);
  expect(result.raw).toBeDefined();
  delete result.raw; // Don't need raw data
  expect(result).toMatchSnapshot();
});

it('should format a thread mention notification', () => {
  let result = formatNotification(MENTION_THREAD_FIXTURE, USER_ID);
  expect(result.raw).toBeDefined();
  delete result.raw; // Don't need raw data
  expect(result).toMatchSnapshot();
});
