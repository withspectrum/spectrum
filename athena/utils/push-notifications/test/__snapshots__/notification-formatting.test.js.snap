// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should format a channel creation notification 1`] = `
Object {
  "body": "\\"New channel3\\" and \\"Test test 1,2,3\\"",
  "title": "2 new channels were created in Whaddup",
}
`;

exports[`should format a community invite notification 1`] = `
Object {
  "data": Object {
    "href": "/whaddup",
  },
  "title": "Max Stoiber invited you to join their community, Whaddup",
}
`;

exports[`should format a direct message creation notification 1`] = `
Object {
  "body": "Max Stoiber: Hey man what's up
Max Stoiber: Testin",
  "data": Object {
    "href": "/messages/7e37e582-3a06-47c9-b799-9a9f7f330ae4",
  },
  "title": "New replies in a direct message thread",
}
`;

exports[`should format a message creation notification 1`] = `
Object {
  "body": "Max Stoiber: Testing",
  "data": Object {
    "href": "/thread/ef021b3b-c520-4133-8749-e96252eff5fc",
  },
  "title": "New thread by @mxstbr (1 new reply)",
}
`;

exports[`should format a message mention notification 1`] = `
Object {
  "body": "\\"Test @mxstbr\\"",
  "data": Object {
    "href": "/thread/2182d960-5970-46e0-801f-ab0b8e02cfbc",
  },
  "title": "Max Stoiber mentioned you in in Show us your Dangerfile",
}
`;

exports[`should format a reaction creation notification 1`] = `
Object {
  "body": "Test 1",
  "data": Object {
    "href": "/thread/2f8087cf-b74d-4587-8ef6-c1b73a751953",
  },
  "title": "Max Stoiber liked your reply",
}
`;

exports[`should format a thread creation notification 1`] = `
Object {
  "body": "\\"Bla\\"",
  "data": Object {
    "href": "/thread/4fbdc540-2ad9-4d6b-8404-736e658be9f9",
  },
  "title": "A new thread was published in General",
}
`;

exports[`should format a thread mention notification 1`] = `
Object {
  "body": "\\"Testing\\"",
  "data": Object {
    "href": "/thread/f40e1d78-10e4-484a-b07d-7e7c88afd167",
  },
  "title": "Max Stoiber mentioned you in in your thread Testing",
}
`;

exports[`should format a user joined community notification 1`] = `
Object {
  "data": Object {
    "href": "/test",
  },
  "title": "Max Stoiber joined Test",
}
`;
